[WIP] Manage provisioning profiles locally

mbogh opened the pull request:
When you are added to 10+ iOS/OSX development teams and add those teams to Xcode you end up with a crazy amount of provisioning profiles. Currently I have 413 installed of which 124 are expired :disappointed:.

So I propose to add `sigh manage (list|cleanup)` for managing provisioning profiles locally.

Please bare with me since it is my first go at ruby :smile_cat: 
### Todo
- [ ] Update README.md
- [x] `sigh manage` to list all installed provisioning profiles.
- [x] `sigh manage -e` to delete expired provisioning profiles.
- [x] `sigh manage -p PATTERN` to delete provisioning profiles matching a pattern.
- [ ] tests?

Commit:
intiial: sigh manage

Commit:
intiial: sigh manage

Commit:
intiial: sigh manage

Commit:
Listing all profiles in red/green.

Commit:
Listing all profiles in red/green.

Commit:
Listing all profiles in red/green.

Commit:
Merge branch 'manage' of github.com:mbogh/sigh into manage

Commit:
Clean up, rebase aint my friend yet

Commit:
Clean up, rebase aint my friend yet

Commit:
Moved profile loading to seperate method and added argument parsing.

Commit:
Moved profile loading to seperate method and added argument parsing.

Commit:
and cleanup is in place

Commit:
and cleanup is in place

KrauseFx commented:
Took a first glance, looks pretty good.

Thanks @mbogh for working on this. Let me know if you need any help or have questions :+1: 

Also, write a comment if there is something to review, so I get a notification.

Thanks :rocket:

Commit:
Removed unnecessary "still"

Commit:
Removed unnecessary "still"

Commit:
load_profiles now returns them sorted.

Commit:
load_profiles now returns them sorted.

Commit:
Better looking output for sigh manage list

Commit:
Better looking output for sigh manage list

Commit:
A little more sugar for `list`

Commit:
A little more sugar for `list`

mbogh commented:
HI @KrauseFx 
I've added the following to options to options.rb

``` ruby
FastlaneCore::ConfigItem.new(key: :clean_team_provisioning,
                                     short_option: "-q",
                                     env_name: "SIGH_CLEAN_TEAM_PROVISIONING",
                                     description: "Remove non-wildcard iOS Team Provisioning e.g. \"iOSTeam Provisioning Profile: *\" survives and \"iOSTeam Provisioning Profile: com.company.app\" moves to .Trash",
                                     optional: true),
        FastlaneCore::ConfigItem.new(key: :clean_custom_pattern,
                                     short_option: "-p",
                                     env_name: "SIGH_CLEAN_CUSTOM_PATTERN",
                                     description: "Remove any provisioning profiles that matches the pattern. Takes precedence over -q",
                                     optional: true)
```

and I want to do something like this in manager.rb
``` ruby
pattern = nil
pattern = /iOS\ {0,1}Team Provisioning Profile: \w/ if Sigh.config[:clean_team_provisioning]
pattern = /#{Sigh.config[:clean_custom_pattern]}/ if Sigh.config[:clean_custom_pattern]
```

But I'm getting the following error:
```
/Users/morten/Github/sigh/lib/sigh/manager.rb:86:in `cleanup_profiles': undefined method `[]' for nil:NilClass (NoMethodError)
	from /Users/morten/Github/sigh/lib/sigh/manager.rb:14:in `run'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/sigh-0.4.7/bin/sigh:55:in `block (2 levels) in run'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/command.rb:179:in `call'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/command.rb:179:in `call'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/command.rb:154:in `run'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/runner.rb:421:in `run_active_command'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/runner.rb:81:in `run!'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/commander-4.3.0/lib/commander/delegates.rb:15:in `run!'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/sigh-0.4.7/bin/sigh:61:in `run'
	from /Users/morten/.rvm/gems/ruby-2.2-head/gems/sigh-0.4.7/bin/sigh:66:in `<top (required)>'
	from /Users/morten/.rvm/gems/ruby-2.2-head/bin/sigh:23:in `load'
	from /Users/morten/.rvm/gems/ruby-2.2-head/bin/sigh:23:in `<main>'
	from /Users/morten/.rvm/gems/ruby-2.2-head/bin/ruby_executable_hooks:15:in `eval'
	from /Users/morten/.rvm/gems/ruby-2.2-head/bin/ruby_executable_hooks:15:in `<main>'
```

mbogh commented:
Never mind, figured it out :)

Commit:
Added commandline options to cleanup. Also ask yes/no before deleting.

Commit:
Added commandline options to cleanup. Also ask yes/no before deleting.

Commit:
Added soon to expire to the list.

Commit:
Added soon to expire to the list.

mbogh commented:
@KrauseFx what is the best way of providing example usage?

Like below

```
$ sigh help manage
  manage

  Usage: sigh manage

  Manage installed provisioning profiles on your system.

  Options:
    -e, --clean_expired  Remove all expired provisioning profiles. 
    -p, --clean_pattern STRING Remove any provisioning profiles that matches the pattern.

  Examples:
  	Remove non-wildcard iOS Team Provisioning e.g. "iOSTeam Provisioning Profile: *" survives and "iOSTeam Provisioning Profile: com.company.app" moves to .Trash
  	sigh manage -p "iOS\ {0,1}Team Provisioning Profile: \w"
```

Commit:
sigh manage simplification

`sigh manage` now lists profiles
`sigh manage -e` cleans expired
`sigh manage -p PATTERN` cleans profiles matching the pattern.

Commit:
sigh manage simplification

`sigh manage` now lists profiles
`sigh manage -e` cleans expired
`sigh manage -p PATTERN` cleans profiles matching the pattern.

KrauseFx commented:
Offtopic: https://github.com/KrauseFx/sigh/pull/72

You might want to check out what I'm working on: I plan on replacing the `poltergeist` code with plain HTTP requests.

Commit:
Added example to sigh help manage

Commit:
Added example to sigh help manage

mbogh commented:
@KrauseFx for now I don't think manage is supposed to delete the profiles on the developer portal...?

Commit:
Kindly inform the user the number of profiles

Commit:
Kindly inform the user the number of profiles

Commit:
Merge branch 'manage' of github.com:mbogh/sigh into manage

KrauseFx commented:
What happened?

mbogh commented:
Brain malfuction in understanding git rebase, no worries a new PR is imminent.
